lib/auth/sessions.go

```diff
@@ -102,6 +106,64 @@ func (s *Server) CreateAppSession(ctx context.Context, req services.CreateAppSes
 	return session, nil
 }
 
+// UpsertAppSession saves the provided app session for the specified user.
+func (s *Server) UpsertAppSession(ctx context.Context, session types.WebSession, identity tlsca.Identity, checker services.AccessChecker) error {
```

Comment:
We now have `CreateAppSession` and `UpdateAppSession` that do fairly different things. If they can't be consolidated, maybe they should be named differently to indicate to the reader what/how they're being used.